OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -182,44 +182,54 @@ public override bool Tick(Actor self)
 
 			var procPos = procLoc.HasValue ? (WPos?)self.World.Map.CenterOfCell(procLoc.Value) : null;
 			var harvPos = self.CenterPosition;
+			var dirPenalty = harvInfo.ResourceRefineryDirectionPenalty;
 
 			// Find any harvestable resources:
 			List<CPos> path;
-			using (var search = PathSearch.Search(self.World, mobile.Locomotor, self, BlockedByActor.Stationary, loc =>
-					domainIndex.IsPassable(self.Location, loc, mobile.Locomotor) && harv.CanHarvestCell(self, loc) && claimLayer.CanClaimCell(self, loc))
-				.WithCustomCost(loc =>
+			using (var search = PathSearch.Search(self.World, mobile.Locomotor, self,
+					BlockedByActor.Stationary,
+					loc => domainIndex.IsPassable(self.Location, loc, mobile.Locomotor)
+						&& harv.CanHarvestCell(self, loc)
+						&& claimLayer.CanClaimCell(self, loc)))
+			{
+				search.WithCustomCost(loc =>
```

Comment:
This relies on an unintuitive (and one could reasonably argue buggy) implementation detail of `.WithCustomCost`: that it modifies the pathsearch it was given in addition to returning a new one.

Can you please rewrite this so that the following code is using the return value from this method, or if there is some important reason that i'm missing to not do that introduce a new `SetCustomCost()` method that makes the intended behaviour clear.